### PR TITLE
Skip moniker range config validation for reference

### PR DESCRIPTION
**Pull request by live1206**
[AB#195160](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/195160)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5712)

### CODE REVIEW

**Comment by mingwli** [comment]
> ⚠[docs-help-pr](https://github.com/OPS-E2E-PPE/docs-help-pr)(00:00:18.6557215, 584 more diff)

Will be fine after a commit ci build

File: src/docfx/config/ops/OpsConfigAdapter.cs [hunk 1]
```diff
@@ -129,6 +129,7 @@ private async Task<string> GetBuildConfig(Uri url)
                     $"{MetadataSchemaApi}{metadataServiceQueryParams}",
                 },
                 xref = xrefMaps,
+                isReference = docset.use_template,
```
Comment (yufeih) [inline src/docfx/config/ops/OpsConfigAdapter.cs:132]: What if `use_template` translate to sth. like `monikerRange { '/': '*' }'` ? I didn't quite like this `isReference` flag.

Comment (live1206) [reply src/docfx/config/ops/OpsConfigAdapter.cs:132]: #5714 to support `*` in moniker range parser

Comment (live1206) [reply src/docfx/config/ops/OpsConfigAdapter.cs:132]: @928PJY had a concern that, if the user misuse `*` a non-versioning repo, then every article will output `monikers` with all the monikers within Docs, which is more than 1000

Comment (yufeih) [reply src/docfx/config/ops/OpsConfigAdapter.cs:132]: So `*` only be used for checking if it is inside a valid moniker range?

Comment (live1206) [reply src/docfx/config/ops/OpsConfigAdapter.cs:132]: how to limit the usage?

Comment (yufeih) [reply src/docfx/config/ops/OpsConfigAdapter.cs:132]: A better name should suffice.

File: src/docfx/build/moniker/MonikerProvider.cs [hunk 2]
```diff
@@ -108,26 +110,33 @@ public MonikerProvider(Config config, BuildScope buildScope, MetadataProvider me
...
+                var (intersectionError, intersection) = GetMonikerIntersection(metadata, configMonikerRange, configMonikers, fileMonikers, _config.SkipMonikerValidation);
                 errors.AddIfNotNull(intersectionError);
                 return (errors, intersection);
             }
 
             return (errors, configMonikers);
         }
 
-        private static (Error?, string[]) GetMonikerIntersection(UserMetadata metadata, SourceInfo<string?> configMonikerRange, string[] configMonikers, string[] fileMonikers)
+        private static (Error?, string[]) GetMonikerIntersection(UserMetadata metadata, SourceInfo<string?> configMonikerRange, string[] configMonikers, string[] fileMonikers, bool isReference)
```
Comment (yufeih) [inline src/docfx/build/moniker/MonikerProvider.cs:128]:
```suggestion
        private static (Error?, string[]) GetMonikerIntersection(UserMetadata metadata, SourceInfo<string?> configMonikerRange, string[] configMonikers, string[] fileMonikers, bool skipMonikerValidation)
```

**Pull request merged**